addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/rest/NestStreamingRestClient.java

```diff
@@ -54,18 +54,22 @@
     private final Logger logger = LoggerFactory.getLogger(NestStreamingRestClient.class);
 
     private final List<NestStreamingDataListener> listeners = new CopyOnWriteArrayList<>();
-    private final EventSource eventSource;
     private final Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").create();
     private final ScheduledExecutorService scheduler;
 
+    private String accessToken;
     private ScheduledFuture<?> checkConnectionJob;
     private boolean connected;
+    private EventSource eventSource;
     private long lastEventTimestamp;
     private TopLevelData lastReceivedTopLevelData;
+    private String redirectUrl;
 
     public NestStreamingRestClient(String accessToken, String redirectUrl, ScheduledExecutorService scheduler) {
-        eventSource = createEventSource(accessToken, redirectUrl);
+        this.accessToken = accessToken;
+        this.redirectUrl = redirectUrl;
         this.scheduler = scheduler;
+        eventSource = createEventSource(accessToken, redirectUrl);
```

Comment:
As accessToken and redirectUrl are now members, is there a need to pass them into createEventSource at all now? - it could be using the members directly.

Is the idea to leave this for dependency injection, so that it could make unit testing easier in the future of this class?

Reply:
I did see it was possible to remove the parameters but I did not have any strong feelings about keeping or removing them. So I left them alone. :-)

Reply:
The parameters are now gone so there is less code and any possible confusion about what data is used where.